crates/sim2h/src/lib.rs

```diff
@@ -411,15 +416,18 @@ pub struct Sim2h {
     sim2h_handle: Sim2hHandle,
     connection_count: usize,
     metric_gen: MetricsTimerGenerator,
+    tracer: ht::Tracer,
```

Comment:
This not being in an option means there is no way to tell if tracing is enabled. This means if there's anything that does have a cost that is done in tracing it will happen all the time

Reply:
feel free to make it Option as we discussed, to avoid unwrapping the opaque message earlier than we need to for times when we're not even doing tracing